### PR TITLE
add log-dir to new dashboard

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?

Add log-dir to new dashboard command to guarantee `dashboard.log` located together with other logs.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/services.py [hunk 1]
```diff
@@ -1053,6 +1057,8 @@ def start_dashboard(require_dashboard,
         f"--redis-address={redis_address}",
         f"--temp-dir={temp_dir}",
     ]
+    if append_log_dir:
+        command += [f"--log-dir={append_log_dir}"]
```
Comment (fyrestone) [inline python/ray/services.py:1061]: Is it better to set the `log_dir` to None when `"RAY_USE_NEW_DASHBOARD" not in os.environ`? Then we can:
``` python
if log_dir:
    command += [f"--log-dir={log_dir}"]
```

Comment (SongGuyang) [reply python/ray/services.py:1061]: LGTM

File: python/ray/services.py [hunk 2]
```diff
@@ -992,6 +992,7 @@ def start_dashboard(require_dashboard,
                     host,
                     redis_address,
                     temp_dir,
+                    log_dir,
```
Comment (richardliaw) [inline python/ray/services.py:995]: Can you please name this as `logdir`?

Comment (SongGuyang) [reply python/ray/services.py:995]: OK.

**Comment by SongGuyang** [comment]
@richardliaw @mfitton Can we merge this PR?

**Comment by richardliaw** [comment]
lgtm, @mfitton should review as this is Dashboard code.

**Review by mfitton — APPROVED** [review]

**Comment by mfitton** [comment]
Hey, sorry, I've been heads down on some other work so hadn't seen this, but it looks good. Just make sure the failing tests are not related please.

**Pull request merged**